services/paymaster/src/api.rs

```diff
@@ -405,24 +429,76 @@ struct SponsorPubkeyQuery {
     #[serde(default)]
     /// Domain to request the sponsor pubkey for
     domain: Option<String>,
+    #[serde(default)]
+    #[param(value_type = String)]
+    /// Index of the sponsor to select. Can be a number or the string "autoassign" which may give you a different sponsor every time
+    index: Option<IndexSelector>,
```

Comment:
what is the use case where someone would want `autoassign`? i don't understand why anyone would want this, instead of just querying different indices manually